content/articles/object-based-image-analysis-using-matlab/index.md

```diff
@@ -0,0 +1,200 @@
+
+### Obtaining boundary by erosion
+In this method, some pixels from a binarized are removed, the eroded part is then subtracted from the main image containing the main object. The remaining part is the boundary of the object. Erosion of image is done using the function `imerode` with a specified `strel` length. The following codes and step by step illustrations of this method;
+```matlab
+i = imread(plate.PNG); %importing the image to workspace
+figure,imshow(i)
+```
+![Imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_seven.jpg)
+```matlab
+i_gray = rgb2gray(i); % converting to grayscale image
+figure,imshow(i_gray)
+```
+![Grayscale version of the imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_eight.jpg)
+```matlab
+i2 = i_gray< 250; %making a black and white image
+figure, imshow (i2)
+```
+![Black and white image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_nine.jpg)
+
+```matlab
+se1 = strel('disk',80); % specifying strel length
+i_eroded = imerode(i2,se1); % erosion process
+figure, imshow(i_eroded)
+```
+![Eroded image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_ten.jpg)
+```matlab
+i_boundary = i2-i_eroded; %subtracting the eroded part of the image
+figure , imshow(i_boundary)
+```
+![Boundary of the object by erosion method](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_eleven.jpg)
+
+### Obtaining boundary by dilation
+Dilation increases the size of the object boundary by adding some pixels to the boundary of the image, the number of pixels added depends on the size and the shape of the object being processed. When the binarized version of the original image is subtracted from the dilated image, the resultant image is the boundary of the targeted object. Dilation of the image is done using the function `imdilate` with a specified `strel` length on a binary image. The output image depends on the original image's contrast and sharpness of pixels. The following are the codes used in this method.
+```matlab
+i = imread('plate.PNG'); %importing an image
+figure,imshow(i)
+```
+![Imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_twelve.jpg)
+
+```matlab
+i2 = rgb2gray(i); %converting to a grayscale image
+figure,imshow(i2)
+```
+![Grayscale version of the imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_thirteen.jpg)
+
+```matlab
+bw1 = imbinarize(i2); % binarizing image
+figure,imshow(bw1)
+```
+![Binary image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_fourteen.jpg)
+
+```matlab
+se = strel('disk',5); % specyfying strel lenght
```

Comment:
Have you explained all these snippets? Kindly explain all codes in this article!!